docs/coding/recipes/balance-invariant-transfers.md

```diff
@@ -18,11 +26,11 @@ This recipe requires three accounts:
   [`flags.credits_must_not_exceed_debits`](../../reference/account.md#flagscredits_must_not_exceed_debits)
   set.
 
-| Id | Debit Account | Credit Account | Amount | Pending Id |                                               Flags |
-| -: | ------------: | -------------: | -----: | ---------: | --------------------------------------------------: |
-|  1 |        Source |    Destination |    123 |          - | [`linked`](../../reference/transfer.md#flagslinked) |
-|  2 |   Destination |        Control |      1 |          - | [`linked`](../../reference/transfer.md#flagslinked), [`pending`](../../reference/transfer.md#flagspending), [`balancing_debit`](../../reference/transfer.md#flagsbalancing_debit) |
-|  3 |             - |              - |      0 |          2 | [`void_pending_transfer`](../../reference/transfer.md#flagsvoid_pending_transfer) |
+| Id | Debit Account | Credit Account |       Amount | Pending Id |                                               Flags |
+| -: | ------------: | -------------: | -----------: | ---------: | --------------------------------------------------: |
+|  1 |        Source |    Destination |    123       |          - | [`linked`](../../reference/transfer.md#flagslinked) |
+|  2 |   Destination |        Control | `AMOUNT_MAX` |          - | [`linked`](../../reference/transfer.md#flagslinked), [`pending`](../../reference/transfer.md#flagspending), [`balancing_debit`](../../reference/transfer.md#flagsbalancing_debit) |
```

Comment:
I think this was already correct with `amount=1`. (`AMOUNT_MAX` works as well, but I think `amount=1` is more clear about the intent.)

The second transfer succeeds if and only if its amount is reduced to `0` (by `balancing_debit`). If the second transfer's final amount is nonzero, then the control account would end up with `debits=0 credits={amount}` (since it initially had `0` credits), which would break its `credits_must_not_exceed_debits` invariant.

See also https://github.com/tigerbeetle/tigerbeetle/blob/444f65b4ec074a36b6030c872a486806bb4521f4/src/state_machine_tests.zig#L1794-L1796 for a test which implements this pattern.